src/EnergyPlus/HeatRecovery.cc

```diff
@@ -1494,10 +1502,9 @@ namespace HeatRecovery {
                 }
             }
 
-            if ((((CompanionCoilType_Num == DataHVACGlobals::CoilDX_CoolingSingleSpeed) ||
-                  (CompanionCoilType_Num == DataHVACGlobals::Coil_CoolingAirToAirVariableSpeed)) &&
-                 (CompanionCoilIndex > 0)) ||
-                ((CompanionCoilType_Num == DataHVACGlobals::CoilDX_Cooling) && (CompanionCoilIndex > -1))) {
+            if ((CompanionCoilIndex > -1) && ((CompanionCoilType_Num == DataHVACGlobals::CoilDX_CoolingSingleSpeed) ||
+                                              (CompanionCoilType_Num == DataHVACGlobals::Coil_CoolingAirToAirVariableSpeed) ||
+                                              (CompanionCoilType_Num == DataHVACGlobals::CoilDX_Cooling))) {
```

Comment:
Rearrainged conditionals using new default -1 instead of looking at > 0 for old coils and -1 for new coil.